rust/hyperlane-base/src/server/server.rs

```diff
@@ -0,0 +1,113 @@
+use axum::{
+    http::{Response, StatusCode},
+    routing::get,
+    Router,
+};
+use hyper::Body;
+use prometheus::{Encoder, Registry};
+use std::{net::SocketAddr, sync::Arc};
+use tokio::task::JoinHandle;
+use tracing::warn;
+
+/// A server that serves agent-specific routes
+pub struct Server {
```

Comment:
nit: can use the `#[derive(new)]` macro here to avoid implementing `new` yourself